Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -2481,7 +2487,10 @@ - (void)_updateVisibleContentRects
     CGFloat scaleFactor = contentZoomScale(self);
     CGRect unobscuredRect = UIEdgeInsetsInsetRect(self.bounds, computedContentInsetUnadjustedForKeyboard);
     CGRect unobscuredRectInContentCoordinates = _perProcessState.frozenUnobscuredContentRect ? _perProcessState.frozenUnobscuredContentRect.value() : [self convertRect:unobscuredRect toView:_contentView.get()];
+    CGPoint unobscuredRectInContentCoordinatesOrigin = unobscuredRectInContentCoordinates.origin;
     unobscuredRectInContentCoordinates = CGRectIntersection(unobscuredRectInContentCoordinates, [self _contentBoundsExtendedForRubberbandingWithScale:scaleFactor]);
```

Comment:
Does FloatRect's intersection behavior do what we want? If so, you could consider using that instead of CGRectIntersection and this little dance

Reply:
@hortont424 what do you mean exactly? The rects being intersected are `CGRect`s, and they can only be intersected one way as far as I know?

Reply:
FloatRect has:
    WEBCORE_EXPORT FloatRect(const CGRect&);
    WEBCORE_EXPORT operator CGRect() const;
so there's nothing stopping you from using it :)

Reply:
(but I don't know it's actually saner! you should check)

Reply:
I mean like what benefit does it have over just using the CGRect intersection like I am currently?

Reply:
It looks like it's not good enough -- it collapses to {0, 0, 0, 0}, not {x, y, 0, 0}

Reply:
so do you mean like
```
    WebCore::FloatRect unobscuredRectInContentCoordinates = WebCore::FloatRect(_perProcessState.frozenUnobscuredContentRect ? _perProcessState.frozenUnobscuredContentRect.value() : [self convertRect:unobscuredRect toView:_contentView.get()]);
    unobscuredRectInContentCoordinates.intersect([self _contentBoundsExtendedForRubberbandingWithScale:scaleFactor]);
    ```